.github/actions/javascript/getDeployPullRequestList/getDeployPullRequestList.ts

```diff
@@ -1,13 +1,81 @@
 import * as core from '@actions/core';
 import * as github from '@actions/github';
+import type {RestEndpointMethodTypes} from '@octokit/plugin-rest-endpoint-methods/dist-types/generated/parameters-and-response-types';
 import {getJSONInput} from '@github/libs/ActionUtils';
 import GithubUtils from '@github/libs/GithubUtils';
 import GitUtils from '@github/libs/GitUtils';
 
+type WorkflowRun = RestEndpointMethodTypes['actions']['listWorkflowRuns']['response']['data']['workflow_runs'][number];
+
+/**
+ * This function checks if a given release is a valid baseTag to get the PR list with `git log baseTag...endTag`.
+ *
+ * The rules are:
+ *     - production deploys can only be compared with other production deploys
+ *     - staging deploys can be compared with other staging deploys or production deploys.
+ *       The reason is that the final staging release in each deploy cycle will BECOME a production release.
+ *       For example, imagine a checklist is closed with version 9.0.20-6; that's the most recent staging deploy, but the release for 9.0.20-6 is now finalized, so it looks like a prod deploy.
+ *       When 9.0.21-0 finishes deploying to staging, the most recent prerelease is 9.0.20-5. However, we want 9.0.20-6...9.0.21-0,
+ *       NOT 9.0.20-5...9.0.21-0 (so that the PR CP'd in 9.0.20-6 is not included in the next checklist)
+ */
+async function isReleaseValidBaseForEnvironment(releaseTag: string, isProductionDeploy: boolean) {
+    if (!isProductionDeploy) {
+        return true;
+    }
+    const isPrerelease = (
+        await GithubUtils.octokit.repos.getReleaseByTag({
+            owner: github.context.repo.owner,
+            repo: github.context.repo.repo,
+            tag: releaseTag,
+        })
+    ).data.prerelease;
+    return !isPrerelease;
+}
+
+/**
+ * Was a given platformDeploy workflow run successful on at least one platform?
+ */
+async function wasDeploySuccessful(runID: number) {
+    const jobsForWorkflowRun = (
+        await GithubUtils.octokit.actions.listJobsForWorkflowRun({
+            owner: github.context.repo.owner,
+            repo: github.context.repo.repo,
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            run_id: runID,
+            filter: 'latest',
+        })
+    ).data.jobs;
+    return jobsForWorkflowRun.some((job) => job.name.startsWith('Build and deploy') && job.conclusion === 'success');
```

Comment:
I feel like `Build and deploy` should be a const somewhere and/or is there a better way to filter if we change the name?

Reply:
I don't have any ideas for a better way to check if one of the specific build jobs passed (other than moving them each to their own callable workflow, but that seems like it should be a separate convo)

Reply:
I can add a const

Reply:
added const and a warning next to the job name